shared/actions/chat/attachment.js

```diff
@@ -418,37 +441,44 @@ function* onRetryAttachment({payload: {message}}: ChatGen.RetryAttachmentPayload
     type: previewType || 'Other',
   }
 
-  yield Saga.put(ChatGen.createRemoveOutboxMessage({conversationIDKey: input.conversationIDKey, outboxID}))
-  yield Saga.call(onSelectAttachment, {payload: {input}, type: ChatGen.selectAttachment, error: false})
+  return Saga.sequentially([
+    Saga.put(ChatGen.createRemoveOutboxMessage({conversationIDKey: input.conversationIDKey, outboxID})),
+    Saga.call(onSelectAttachment, {payload: {input}, type: ChatGen.selectAttachment, error: false}),
+  ])
 }
 
-function* onOpenAttachmentPopup(action: ChatGen.OpenAttachmentPopupPayload): SagaGenerator<any, any> {
+function onOpenAttachmentPopup(action: ChatGen.OpenAttachmentPopupPayload) {
   const {message, currentPath} = action.payload
   const messageID = message.messageID
   if (!messageID) {
     throw new Error('Cannot open attachment popup for message missing ID')
   }
 
-  yield Saga.put(
-    putActionIfOnPath(
-      currentPath,
-      navigateAppend([{props: {messageKey: message.key}, selected: 'attachment'}])
+  const actions = []
+  actions.push(
+    Saga.put(
+      putActionIfOnPath(
+        currentPath,
+        navigateAppend([{props: {messageKey: message.key}, selected: 'attachment'}])
+      )
     )
   )
   if (!message.hdPreviewPath && message.filename && message.messageID) {
-    yield Saga.put(ChatGen.createLoadAttachment({messageKey: message.key, loadPreview: false}))
+    actions.push(Saga.put(ChatGen.createLoadAttachment({messageKey: message.key, loadPreview: false})))
   }
+
+  return Saga.sequentially(actions)
 }
 
 function attachmentLoaded(action: ChatGen.AttachmentLoadedPayload) {
   const {payload: {messageKey, path, isPreview}} = action
   if (isPreview) {
-    return Saga.all([
+    return Saga.sequentially([
```

Comment:
Did you find that this mattered?

Reply:
I think in practice with just puts it's essentially the same but mike asked about it and I think making it more explicit is likely better.